src/main/java/com/github/flaxsearch/api/BKDNode.java

```diff
@@ -74,13 +75,21 @@ public BKDNode findParent(BKDNode node, int numDims, int bytesPerDim) {
     }
 
     public boolean contains(BKDNode node, int numDims, int bytesPerDim) {
-        boolean contains = true;
         for (int i = 0; i < numDims; i++) {
             int offset = i * bytesPerDim;
-            contains &= (StringHelper.compare(bytesPerDim, minPackedValue, offset, node.minPackedValue, offset) > 0 &&
-                    StringHelper.compare(bytesPerDim, maxPackedValue, offset, node.maxPackedValue, offset) < 0);
+            int cMin = StringHelper.compare(bytesPerDim, minPackedValue, offset,
+                                            node.minPackedValue, offset);
+            int cMax = StringHelper.compare(bytesPerDim, maxPackedValue, offset,
+                                            node.maxPackedValue, offset);
+            if (cMin > 0 || cMax < 0) return false;
         }
```

Comment:
Can you add a test for this?  I should have put one in there in the first place, really...